### PR TITLE
Feature/milestone 5: Tying calculations module with the components and adding interaction to the buttons

### PR DESCRIPTION
**Pull request by tundeiness**
**The following were achieved in this milestone:**

- App component was implemented as stateful
- App component initialises states with total, next and operation
- handleClick() was implemented and passed to ButtonPanel
- Clickhandler() was passed to each button in the panel
- handleClick returns clickHandler from props

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -9,16 +8,39 @@ class App extends React.Component {
...
+    }
     return (
       <div className="app-wrapper">
-        <Display result={result} />
-        <ButtonPanel />
+        <Display />
+        <ButtonPanel clickHandler={(buttonName) => this.handleClick(buttonName)} />
       </div>
     );
   }
```
Comment (bolah2009) [inline src/components/App.js]: [Optional]

I think these two `return` is unnecessary. When `res` is `undefined` the `defaultProps` is used. returning only the first case will do.
Let's delete the second case and return `res` as `result` props

File: src/logic/operate.js [hunk 2]
```diff
@@ -2,24 +2,33 @@
...
-  } if (operation === 'x') {
+  }
+
+  if (operation === 'x') {
     return first.times(second).toString();
-  } if (operation === '÷') {
+  }
+
+  if (operation === '÷') {
     if (second === 0) {
```
Comment (bolah2009) [inline src/logic/operate.js]: I think it's safer to test if `numTwo` is zero. 

Currently, you haven't handle division by zero properly.

caldiverr0

**Review by bolah2009 — APPROVED** [review]
It's a 👍🏾 from me!

### :heavy_check_mark: Approved!

Check out the comments before merging. :relaxed: 

Good luck with your next projects. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Comment by tundeiness** [comment]
**The following corrections were made to this pull request:**

- Division by 0 was properly handled
- In App.js an extra return was removed.

**Pull request merged**